chore(release): Zebra release v2.2.0 (#9195)
ZcashFoundation/zebra: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org).

## [Zebra 2.2.0](https://github.com/ZcashFoundation/zebra/releases/tag/v2.2.0) - 2025-02-03

> In this release, Zebra fixed a consensus bug in the branch ID of Nu6 transactions, along with important refactors and improvements.

Reviewer: Optional: this seems more alarming than it really is

```suggestion
In this release, Zebra introduced an additional consensus check on the branch ID of Nu6 transactions
(which is currently also checked elsewhere; but we believe it's important to check on its own to protect
against future code changes), along with important refactors and improvements.
```

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org).

## [Zebra 2.2.0](https://github.com/ZcashFoundation/zebra/releases/tag/v2.2.0) - 2025-02-03

In this release, Zebra fixed a consensus bug in the branch ID of Nu6 transactions, along with important refactors and improvements.

### Added

- An index to track spending transaction ids by spent outpoints and revealed nullifiers ([#8895](https://github.com/ZcashFoundation/zebra/pull/8895))

> - Public getter / setter functions to Zebra-RPC::methods.rs ([#9113](https://github.com/ZcashFoundation/zebra/pull/9113))

Reviewer 2: ```suggestion
- Accessor methods to `zebra-rpc` request/response types ([#9113](https://github.com/ZcashFoundation/zebra/pull/9113))
```

- `getblock` RPC method now can return transaction details with verbosity=2 ([#9083](https://github.com/ZcashFoundation/zebra/pull/9083))

> - A missing serialized NU5 blocks to test vectors ([#9098](https://github.com/ZcashFoundation/zebra/pull/9098))

Reviewer 2: Nitpick/Subjective:

```suggestion
- Serialized NU5 blocks to test vectors ([#9098](https://github.com/ZcashFoundation/zebra/pull/9098))
```

### Changed

- Migrated from deprecated `jsonrpc_*` crates to `jsonrpsee` ([#9059](https://github.com/ZcashFoundation/zebra/pull/9059), [#9151](https://github.com/ZcashFoundation/zebra/pull/9151))
- Optimized checks for coinbase transactions ([#9126](https://github.com/ZcashFoundation/zebra/pull/9126))

> - Avoid re-verifying transactions in blocks if those transactions are in the mempool ([#8951](https://github.com/ZcashFoundation/zebra/pull/8951))

Reviewer 2: ```suggestion
- Avoid re-verifying transactions in blocks if those transactions are in the mempool ([#8951](https://github.com/ZcashFoundation/zebra/pull/8951), [#9118](https://github.com/ZcashFoundation/zebra/pull/9118))
```

- Allow transactions spending coinbase outputs to have transparent outputs on Regtest ([#9085](https://github.com/ZcashFoundation/zebra/pull/9085))

### Fixed

- Respond to getblockchaininfo with genesis block when empty state ([#9138](https://github.com/ZcashFoundation/zebra/pull/9138))

> - Avoid a concurrency bug when verifying transactions in blocks that are already present in the mempool ([#9118](https://github.com/ZcashFoundation/zebra/pull/9118))

Reviewer 2: Could we include this as part of "Avoid re-verifying transactions in blocks if those transactions are in the mempool"? (see [next suggestion](https://github.com/ZcashFoundation/zebra/pull/9195#discussion_r1939972701))

```suggestion
```

- Verify consensus branch ID in SIGHASH precomputation ([#9139](https://github.com/ZcashFoundation/zebra/pull/9139))

> - Refactor `getrawtransaction` and RPC error handling ([#9049](https://github.com/ZcashFoundation/zebra/pull/9049))

Reviewer 2: Nitpick: We should focus this on user-visible changes.

Related: Zebra may still need to match zcashd's `getrawtransaction` output when transactions are found on side chains (it currently returns an error if the transaction isn't in the best chain or mempool, zcashd returns the transaction with a height of -1).

```suggestion
- More closely match zcashd RPC errors and `getrawtransaction` RPC behaviour ([#9049](https://github.com/ZcashFoundation/zebra/pull/9049))
```
